sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdChannelAcquisitionContext.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.directconnectivity.rntbd;
+
+import com.azure.cosmos.implementation.apachecommons.lang.tuple.Pair;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.azure.cosmos.implementation.directconnectivity.rntbd.RntbdReporter.reportIssueUnless;
+
+public class RntbdChannelAcquisitionContext {
+    private static final Logger logger = LoggerFactory.getLogger(RntbdChannelAcquisitionContext.class);
+    /**
+     * Track each time the request being added to pending acquisition queue.
+     */
+    @JsonSerialize(using = ToStringSerializer.class)
+    private final List<Instant> pendingTimestamps;
+
+    /**
+     * Track each time the request try to acquire channel.
+     */
+    @JsonSerialize(using = ToStringSerializer.class)
+    private final List<Instant> acquireChannelTimestamps;
```

Comment:
```suggestion
    private final List<Instant> attemptedToAcquireChannelTimestamps;
```

Reply:
changed the representation to timeline format, this class has removed